### PR TITLE
feat!: Update `function` --> `func`

### LINKED ISSUE #31: [Bug] version mismatch
In the last update `cargo_witgen`, didn't update the dependency to the newest version.

https://github.com/bnjjj/witgen/blob/c5fd2e98119dd8bf26b85d1d1c86c1e16b7b8a6d/crates/cargo_witgen/Cargo.toml#L22

I think perhaps it's finally time to invest in a GH action to handle the versioning and publishing.

### PR DESCRIPTION
**Pull request by willemneal**
Also fixes #31.  

Also looked into `cargo-smart-release`.  This should work:

```bash
cargo smart-release witgen --bump minor --no-changelog --update-crates-index
```

Add `--execute` for it to actually do it.

This PR ended up growing in scope.
Updated:
 - wit-parser
 
Added:
- `use` (import statements) and `resource` (impls, however, they only support functions at this time); `interface` (trait, but this type is not well supported by wit yet).
- `impl`'s methods now support special attributes.  e.g. 
```rust
impl Foo {
  #[special_attr]
  pub fn foo(){}
}
```
Becomes
```wit
resource foo {
  ///@special_attr
  foo: func()
}
```

- resolver to pull in wit dependencies from `use` (still need to test multiple depths, currently okay with one)
- clap-cargo-extra, a new crate a made for dealing with cargo CLI and metadata
- k9 testing library which provides snapshotting

I know you already approved, but if you could just leave a comment that should be enough ;-)

### CODE REVIEW

**Comment by willemneal** [comment]
Still need to update to the newest parser.

**Review by bnjjj — APPROVED** [review]

**Comment by willemneal** [comment]
@bnjjj I know it's a big one, but your eyes would be much appreciated!

File: crates/witgen_macro_helper/src/generator.rs [hunk 1]
```diff
@@ -110,36 +104,18 @@ pub fn gen_wit_enum(enm: &ItemEnum) -> Result<String> {
         .variants
         .iter()
         .map(|variant| {
-            let ident = variant.ident.to_string().to_kebab_case();
-            let comment = get_doc_comment(&variant.attrs)?;
+            let ident = wit_ident(&variant.ident)?;
+            let comment = get_doc_comment(&variant.attrs, 1, false)?;
             let variant_string = match &variant.fields {
                 syn::Fields::Named(_named) => {
```
Comment (bnjjj) [inline crates/witgen_macro_helper/src/generator.rs:110]:
```suggestion
                syn::Fields::Named(named) => {
```

File: crates/witgen_macro_helper/src/generator.rs [hunk 2]
```diff
@@ -110,36 +104,18 @@ pub fn gen_wit_enum(enm: &ItemEnum) -> Result<String> {
...
-                                    "{}    {}: {}",
-                                    field_doc,
-                                    field.ident.as_ref().unwrap().to_string().to_kebab_case(),
-                                    ty
-                                )
-                            })
-                        })
-                        .collect::<Result<Vec<String>>>()?
-                        .join(",\n");
+                    let fields = gen_fields(_named.named.iter().collect())?.join(",\n");
```
Comment (bnjjj) [inline crates/witgen_macro_helper/src/generator.rs:111]:
```suggestion
                    let fields = gen_fields(named.named.iter().collect())?.join(",\n");
```

File: crates/witgen_macro_helper/src/generator.rs [hunk 3]
```diff
@@ -250,23 +200,90 @@ pub fn gen_wit_type_alias(type_alias: &ItemType) -> Result<String> {
...
+    Ok(comment)
+}
+
+pub fn gen_wit_import(import: &ItemUse) -> Result<String> {
+    let import_name = match &import.tree {
+        UseTree::Path(UsePath { ident, .. }) => wit_ident(ident)?,
+        UseTree::Name(_) => todo!(),
+        UseTree::Rename(_) => todo!(),
+        UseTree::Glob(_) => todo!(),
+        UseTree::Group(_) => todo!(),
```
Comment (bnjjj) [inline crates/witgen_macro_helper/src/generator.rs:239]: Could you add a message to these todo please ?

**Review by bnjjj — APPROVED** [review]
It looks good overall. Could you just add in the README the current wit version this codebase is supporting please ? It would be easier for our users to discover which features of wit we support

**Review by bnjjj — APPROVED** [review]

**Comment by bnjjj** [comment]
@willemneal published on `0.15.0`

**Pull request merged**